File: app/src/main/java/me/gilo/starter/common/CompletionLiveData.java
package me.gilo.starter.common;


import androidx.annotation.NonNull;
import androidx.lifecycle.LiveData;
import com.google.android.gms.tasks.OnCompleteListener;
import com.google.android.gms.tasks.Task;


public final class CompletionLiveData extends LiveData<Resource<Boolean>> implements OnCompleteListener<Void> {


    public CompletionLiveData() {
        setValue(new Resource<>(Status.LOADING));
    }

    @Override
    public final void onComplete(@NonNull Task<Void> task) {
        if (task.isSuccessful()) {
            setValue(new Resource<>(true));
        } else {
            setValue(new Resource<>(task.getException()));
        }
    }
}

File: app/src/main/java/me/gilo/starter/common/CompletionLiveData.kt
package me.gilo.starter.common


import androidx.lifecycle.LiveData
import com.google.android.gms.tasks.OnCompleteListener
import com.google.android.gms.tasks.Task


class CompletionLiveData : LiveData<Resource<Boolean>>(), OnCompleteListener<Void> {
    init {
        value = Resource(Status.LOADING)
    }

    override fun onComplete(task: Task<Void>) {
        if (task.isSuccessful) {
            value = Resource(true)
        } else {
            value = Resource(task.exception!!)
        }
    }
}
